crates/core/tests/client.rs

```diff
@@ -1,22 +1,137 @@
+            let address : Felt252Wrapper = (*deployed_addresses.first().unwrap()).into();
+            address.try_into()
+        }) else {
+            panic!("returned deploy address from kakarot failed to convert to address")
+        };
+
+        kakarot_client.get_code(opcode_eth_addr, BlockId::Tag(BlockTag::Latest)).await.expect("contract not deployed");
+    }
+
+    #[tokio::test]
+    #[ignore]
+    async fn test_counter() {
+        let starknet_test_sequencer = TestSequencer::start().await;
+
+        let eoa_public = Address::from_slice(&hex::decode(EVM_ADDRESS.trim_start_matches("0x")).unwrap());
+        let eoa_private = H256::from_slice(&hex::decode(EVM_PRIVATE_KEY).unwrap());
+
+        let expected_funded_amount = FieldElement::from_dec_str("100000000000000000").unwrap();
+
+        let deployed_kakarot =
+            deploy_kakarot_system(&starknet_test_sequencer, EVM_ADDRESS, eoa_private, expected_funded_amount).await;
+
+        let (counter_abi, deployed_addresses) = deployed_kakarot
+            .deploy_evm_contract(
+                starknet_test_sequencer.url(),
+                "Counter",
+                // no constructor is conveyed as a tuple
+                (),
+            )
+            .await
+            .unwrap();
+
+        let kakarot_client = KakarotClient::new(StarknetConfig::new(
+            starknet_test_sequencer.url().as_ref(),
+            deployed_kakarot.kakarot,
+            deployed_kakarot.kakarot_proxy,
+        ))
+        .unwrap();
+
+        let deployed_balance = kakarot_client.balance(eoa_public, BlockId::Tag(BlockTag::Latest)).await;
+
+        let deployed_balance = FieldElement::from_bytes_be(&deployed_balance.unwrap().to_be_bytes()).unwrap();
+
+        assert_eq!(deployed_balance, expected_funded_amount);
+
+        let Ok(counter_eth_addr) = ({
+            let address : Felt252Wrapper = (*deployed_addresses.first().unwrap()).into();
+            address.try_into()
+        }) else {
+            panic!("returned deploy address from kakarot failed to convert to address")
+        };
+
+        kakarot_client.get_code(counter_eth_addr, BlockId::Tag(BlockTag::Latest)).await.expect("contract not deployed");
+
+        let inc_selector = counter_abi.function("inc").unwrap().short_signature();
+
+        let nonce = kakarot_client.nonce(eoa_public, BlockId::Tag(BlockTag::Latest)).await.unwrap();
+        let inc_tx =
+            create_raw_ethereum_tx(inc_selector, eoa_private, counter_eth_addr, vec![], nonce.try_into().unwrap());
+        let inc_res = kakarot_client.send_transaction(inc_tx).await.unwrap();
+        kakarot_client.transaction_receipt(inc_res).await.expect("increment receipt failed");
+
+        let count_selector = counter_abi.function("count").unwrap().short_signature();
+        let counter_val =
+            kakarot_client.call_view(counter_eth_addr, count_selector.into(), BlockId::Tag(BlockTag::Latest)).await;
+
+        if let Ok(bytes) = counter_val {
```

Comment:
You can `counter_val.expect(<error msg goes here>)` instead of the if elses imho

Reply:
even just unwrapping here is fine